packages/server/tests/acceptance/estimateGasPrecompile.spec.ts

```diff
@@ -0,0 +1,473 @@
+    );
+
+    let estimateContractSigner2 = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[2].wallet,
+    );
+    const tx2_ft = await estimateContractSigner2.associateTokenExternal(
+      accounts[2].wallet.address,
+      tokenAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    const t1_nft = await estimateContractSigner1.associateTokenExternal(
+      accounts[1].wallet.address,
+      nftAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    const tx2_nft = await estimateContractSigner2.associateTokenExternal(
+      accounts[2].wallet.address,
+      nftAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    await new Promise((r) => setTimeout(r, 2500));
+    //NFT
+    await contract.grantTokenKycExternal(nftAddress, accounts[0].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[1].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, accounts[2].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(nftAddress, prefix + EstimatePrecompileContractAddress, { gasLimit: 500_000 });
+
+    //Fungible
+    await contract.grantTokenKycExternal(tokenAddress, accounts[0].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[1].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, accounts[2].wallet.address, { gasLimit: 500_000 });
+    await contract.grantTokenKycExternal(tokenAddress, prefix + EstimatePrecompileContractAddress, {
+      gasLimit: 500_000,
+    });
+  });
+
+  const baseGasCheck = (estimatedGasValue, expectedValue: number) => {
+    // handle deviation of 20%
+    expect(Number(estimatedGasValue)).to.be.lessThan(expectedValue * 1.4);
+  };
+
+  //EGP-001
+  it('should call estimateGas with associate function for fungible token', async function () {
+    const tx = await contract.associateTokenExternal.populateTransaction(signers[0].address, tokenAddress);
+    const estimateGasResponse = await relay.call(RelayCalls.ETH_ENDPOINTS.ETH_ESTIMATE_GAS, [tx]);
+    baseGasCheck(estimateGasResponse, 786166);
+  });
+
+  //EGP-002
+  it('should call estimateGas with associate function for NFT', async function () {
+    const tx = await contract.associateTokenExternal.populateTransaction(signers[0].address, nftAddress);
+    const estimateGasResponse = await relay.call(RelayCalls.ETH_ENDPOINTS.ETH_ESTIMATE_GAS, [tx]);
+    baseGasCheck(estimateGasResponse, 786166);
+  });
+
+  //EGP-003
+  it('should call estimateGas with dissociate token function without association for fungible token - negative', async function () {
+    const tx = await contract.dissociateTokenExternal.populateTransaction(
+      prefix + EstimatePrecompileContractAddress,
+      tokenAddress,
+    );
+    negativeScenarioVerification(tx, SERVER_ERROR);
```

Comment:
Does `SERVER_ERROR` mean we get status code 500 from the API. If that's the case, this means the test is not doing something correctly, because we don't expect code 500 in a negative scenarios, only 400 that results in `CONTRACT_REVERT_EXECUTED`.
Same comment applies for the tests below.